tensorboard/webapp/widgets/line_chart_v2/sub_view/line_chart_unencapsulated.ts

```diff
@@ -0,0 +1,46 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ViewEncapsulation,
+} from '@angular/core';
+
+/**
+ * Why this weird hack?
+ * The prospective fob area is absolutely positioned over the XAxis.
+ * This prevents hover events from reaching the XAxis.
+ * The .extent-edit-button should appear when the XAxis is hovered.
+ */
+const prospectiveFobAreaHack = `
+.line-chart:has(.horizontal-prospective-area:hover) {
```

Comment:
By the way I only coincidentally learned today (at goog.web) that :has is a new property and isn't yet fully supported in Firefox. It seems like support is coming soon, though. Just wanted to make sure you're aware. I think we can continue to rely on this and we'll just have to wait a little longer for the fix to work in Firefox.

Reply:
Hmm, I was not aware that `:has` was not fully implemented in FireFox and yeah, this actually does not work there. Given that this is a pretty small issue and a fix is coming soon I don't think that should be a blocker though.
On a side note I'm planning on watching the replay of goog.web later